adoc/extensions/sycl_khr_group_interface.adoc

```diff
@@ -0,0 +1,495 @@
+work-items as [code]#sg#.
+
+'''
+
+.[apititle]#sub_group conversion operator#
+[source,role=synopsis,id=api:khr-group-interface-sub-group-conversion-operator]
+----
+operator sycl::sub_group() const noexcept;
+----
+
+_Returns_: A [code]#sycl::sub_group# representing the same collection of
+work-items as this [code]#sub_group#.
+
+'''
+
+.[apidef]#khr::sub_group::max_size#
+[source,role=synopsis,id=api:khr-group-interface-sub-group-max-size]
+----
+constexpr size_type max_size() const noexcept;
+----
+
+_Returns_: The maximum number of work-items permitted in any <<sub-group>> for
+the executing kernel.
+
+{note}There is no guarantee that any sub-group within the work-group contains
+the maximum number of work-items.{endnote}
+
+_Remarks_: The value returned by this function must reflect the value passed to
+the [code]#reqd_sub_group_size# attribute, if present.
+If no such attribute is present, the value returned is determined by the
+<<device-compiler>>.
+
+'''
+
+[[sec:khr-group-interface-work_item]]
+== [code]#work_item# class
+
+The [code]#work_item# class template encapsulates all functionality required to
+represent a single <<work-item>> within a specific <<group>> of work-items.
+
+The mechanism used to determine the calling work-item's position within a given
+group of work-items is implementation-defined, and there is subsequently no way
+for a user to construct arbitrary instances of the [code]#work_item# class
+template.
+Instances of the [code]#work_item# class template can only be acquired from a
+call to [api]#khr::get_item#.
+
+The SYCL [code]#work_item# class template provides common by-value semantics
+(see <<sec:byval-semantics>>) and the common group interface (see
+<<sec:khr-group-interface-common>>).
+
+[source,role=synopsis]
+----
+namespace sycl::khr {
+
+template <typename ParentGroup>
+class work_item {
+
+ public:
+  using id_type = typename ParentGroup::id_type;
+  using linear_id_type = typename ParentGroup::linear_id_type;
+  using range_type = typename ParentGroup::range_type;
+  using extents_type = /* extents of all 1s with ParentGroup's index type */; // C++23
+  using size_type = typename ParentGroup::size_type;
+  static constexpr int dimensions = ParentGroup::dimensions;
+  static constexpr memory_scope fence_scope = memory_scope::work_item;
+
+  /* -- common by-value interface members -- */
+
+  /* -- common group interface members -- */
+
+};
+
+} // namespace sycl::khr
+----
+
+[[sec:khr-group-interface-example]]
+== Example
+
+The example below demonstrates the usage of this extension.
+
+[source,,linenums]
+----
+#include <sycl/sycl.hpp>
+#include <iostream>
+#include <numeric>
+#include <algorithm>
+using namespace sycl; // (optional) avoids need for "sycl::" before SYCL name
+
+constexpr size_t N = 1024;
+constexpr size_t M = 256;
+
+int main() {
+
+  queue q;
+
+  int* in = malloc_shared<int>(N * M, q);
+  int* out = malloc_shared<int>(N, q);
```

Comment:
Is this code UB because there is no `free`? :thinking:

Reply:
Fixed in f4300546d20b7a65378d91b4e0737187d364d6bf.